app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,41 @@
+class CommentsController < ApplicationController
+  before_action :set_user_and_post, only: %i[new create]
+
+  def new
+    store_referer
+    @comment = Comment.new
+    @post = Post.find(params[:post_id])
+  end
+
+  def create
+    @comment = @post.comments.build(comment_params.merge(author_id: current_user.id))
+
+    if @comment.save
+      flash[:success] = 'Comment saved successfully'
+      redirect_back_or_default(user_post_comments_path(@user, @post))
```

Comment:
- [ ] [REQUIRED] Great efforts @evansnyamekye ,
However, I am unable to create a comment at the moment. The issue is related to the redirect_back_or_default method. The error message indicates that the :user_id key is missing when trying to generate a URL in the create action of the CommentsController.
<img width="1107" alt="Screenshot 2023-12-15 at 15 27 38" src="https://github.com/evansnyamekye/blog-app/assets/58692590/96a302a8-ae87-4029-96a6-a60e3a3d2fdb">

### To resolve this issue
- The @user variable is not defined in your create action, and that's why the :user_id key is missing. You should use current_user instead

``` ruby
redirect_to new_user_post_comment_path(current_user, @post)
```

Reply:
Hi, @Whoistolu. The issue with the comment has been resolved. kindly find below a video.  

https://github.com/evansnyamekye/blog-app/assets/60971417/c61e36b2-dc36-4582-aac4-19e9892822f0